### ISSUE TITLE
mv candidates in BMA for multi-ref prediction should only include the same ref mvs.

### ISSUE THREAD

**Issue by ycho** [xiph/daala#144]
I found this bug in my bi-pred work, and expect fixing this bug will help improving the lost coding gain of muti-ref as well!
Specifically, inside the function od_mv_est_init_mv(), whenever it construct mv candidates from spatial or temporal neighbors, it should only consider the same ref mvs.

**Comment by tdaede** [comment]
This was fixed in 6dcb4ce8902d56d3ff139119e8d39b564a0ee68a.

**Comment by ycho** [comment]
Yes, I know the mv predictors is fixed before and I also use it.
But I mean when deciding the candidates mvs in BMA, not the predictors, pred[] which returns from get_predictor() function.
So, for ex, cneighbors[ci]->bma_mvs[0][ref], we don't know whether the position cneighbors[ci] will have valid mv for ref == backward prediction (or, golden frame).

**Comment by tdaede** [comment]
Oh okay. Yeah I don't think reducing the # of candidates checked will actually be beneficial, but I'll leave it open if someone wants to try.

**Comment by ycho** [comment]
I hope this can fix my big loss of b-frame (bma only) for some sequences.
FYI, when I applied your fix for get_predictor(), I got +5% for B-frame+bma
only!
So, I hope again excluding diffrent ref mvs will improve a lot.

-yushin

On Mon, Oct 12, 2015 at 11:54 AM, Thomas Daede <notifications@github.com>
wrote:

> Oh okay. Yeah I don't think reducing the # of candidates checked will
> actually be beneficial, but I'll leave it open if someone wants to try.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xiph/daala/issues/144#issuecomment-147490647>.
>



-- 
Thanks!
Yushin
